### ISSUE TITLE
Didn't work for me without some groundwork

### ISSUE THREAD

**Issue by kingboyk** [u3mur4/terracoin#1]
Hello and thanks for the script. I tried it several times but terracoind wouldn't start due to missing dependencies. I deleted the user and user folder your script created, ran all of the apt related commands in [clockuniverse's guide](https://drive.google.com/file/d/0B6wHcTCr1qh3S09rMUI5dmhoNnc/edit), and then ran your script again, this time successfully.

Conclusion: You likely missed out some apt commands. I don't know which ones, unfortunately.

**Comment by u3mur4** [comment]
Hi,

That would be very strange because I tried myself on a new vps before I
shared the guide. Are you using ubuntu 16.04?
(I copied all dependencies from clockuniverse's guide to the setup script.)

Regard,
u3mur4

2017-09-26 17:05 GMT+02:00 kingboyk <notifications@github.com>:

**Comment by kingboyk** [comment]
Yes, an x64 Ubuntu 16.04.03 LTS VPS. Kernel is Linux 2.6.32.

**Comment by u3mur4** [comment]
I am pretty sure you are using some lame vps provider. I tried once again the script on vultr and everything was fine.

Can you show me what is the output of this command?
```
cat /etc/os-release
```

The script installing this libraries: https://github.com/u3mur4/terracoin/blob/master/trc.py#L121
Can you run the following commands and notify me if you get any error?
```bash
add-apt-repository ppa:bitcoin/bitcoin -y
```
```bash
apt-get update
```
```bash
apt-get --assume-yes install git unzip libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-program-options-dev libboost-test-dev libboost-thread-dev software-properties-common libdb4.8-dev libdb4.8++-dev libminiupnpc-dev libevent-dev libzmq3-dev
```

P.S. I have indicated in the guide that I don't support other vps with the exception of vultr or digitalocean.

**Comment by kingboyk** [comment]
You don't actually need to support _me_ (and your PS is noted), but if you're interested in confirming there's no problem with the script let's do it! Otherwise let's just close the issue and move on :)

Lame VPS provider? Probably. Certainly cheap. Masternode is still up though and the secure shell seems responsive.

cat /etc/os-release:
```
NAME="Ubuntu"
VERSION="16.04.3 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.3 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

Other commands:
No errors. Nothing new to install.

**Comment by u3mur4** [comment]
I have written the PS not just for you, but also others who read this issue. If I want to support every differences of the vps providers, that would be to much work. I want to help the community but I don't have that amount of free time to help everyone one by one. I knew that you fixed the error but not everyone is so skillful like you.

I copied the necessary libraries from clockuniverse's guide (what I pasted in my previous answer). It looks like it is all installed in your system so it should work without any other libraries. Can you provide some libraries that you installed in addition? If not I think I can't help and I will close the issue.

Thank you and have a great day!

**Comment by kingboyk** [comment]
Unfortunately I didn't make notes. The best I could do is paste in my command history. Shall I do that or would you prefer to close the issue?

**Comment by u3mur4** [comment]
Do that and if I don't find anything interesting I will close the issues.
Thanks!

**Comment by kingboyk** [comment]
OK, good plan :) Here goes:

   3  apt-get install -y python ; rm trc.py; wget https://raw.githubusercontent.com/u3mur4/terracoin/master/trc.py && python trc.py
    5  sudo apt-get install wajig
    6  ps -a
    7  sudo wajig daily-upgrade
    8  ls
    9  ls -a
   10  top
   11  ls
   12  ls -a
   13  man ps
   14  ps -e
   15  suc -mn1
   16  su -mn1
   17  su - mn1
   18  sudo apt-get install libboost1.54-all-dev
   19  sudo apt-get install libboost1.58-all-dev
   20  sudo apt-get install libdb-dev libdb++-dev
   21  sudo apt-get install libminiupnpc-dev
   22  su - mn1
   23  ln -s /usr/local/lib/libevent-2.0.so.5 /usr/lib/libevent-2.0.so.5
   24  su - mn1
   25  ls /usr/local/lib/libevent-2.0.so.5 /usr/lib/libevent-2.0.so.5
   26  ls -l /usr/local/lib/libevent-2.0.so.5 /usr/lib/libevent-2.0.so.5
   27  rm /usr/local/lib/libevent-2.0.so.5 /usr/lib/libevent-2.0.so.5
   28  sudo apt-get install libevent-2.0.so.5
   29  sudo apt-get install libevent-pthreads-2.0-5
   30  su - mn1
   31  sudo ln -s /usr/local/lib/libevent-2.0.so.5 /usr/lib64/libevent-2.0.so.5
   32  sudo apt-get install libzmq3-dev
   33  su - mn1
   34  updatedb
   35  locate terra
   36  locate terra|more
   37  terracoind -printtoconsole
   38  sudo apt-get install libdb4.8-dev libdb4.8++-dev
   39  sudo apt-get install libdb4.9-dev libdb4.9++-dev
   40  python trc.py
   41  rmuser
   42  userdel mn1
   43  updatedb
   44  locate mn1
   45  rm -frv /mn1
   46  locate mn1
   47  updatedb
   48  locate mn1
   49  rm -frv /home/mn1
   50  updatedb
   51  locate mn1
   52  ls
   53  python trc.py
   54  ps e
   55  top
   56  sudo apt-get install libdb5.3++
   57  export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/usr/local/lib"
   63  su - mn1
   64  sudo apt-get install libdb_cxx-4.8
   65  sudo apt-get install libdb_cxx-4.8.so
   66  locate  libdb_cxx-4.8.so
   67  wget http://download.oracle.com/berkeley-db/db-4.8.30.NC.tar.gz
   68  tar -xzvf db-4.8.30.NC.tar.gz
   69  cd db-4.8.30.NC/build_unix
   70  ../dist/configure --enable-cxx --prefix=/usr
   71  make -j8
   72  sudo checkinstall --fstrans=no --pkgversion=4.8.30 --pgkname=libdb-dev --nodoc -y
   73  userdel mn1
   74  rm -frv /home/mn1
   75  ls
   76  ls -a
   77  sudo apt-get update
   78  sudo apt-get upgrade
   79  sudo apt-get dist-upgrade
   80  sudo apt-get install libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-program-options-dev libboost-thread-dev
   81  sudo apt-get install software-properties-common
   82  sudo add-apt-repository ppa:bitcoin/bitcoin
   83  sudo apt-get update
   85  apt-get install libdb4.8-dev libdb4.8++-dev
   86  sudo apt-get install libminiupnpc-dev
   87  sudo apt-get install libevent-dev
   88  sudo apt-get install libzmq3-dev
   89  terracoind -printtoconsole
   90  ls
   91  ls -a
   92  more /etc/crontab
   93  crontab
   94  man crontab
   95  crontab -l
   96  crontab -l -u mn1
   97  ls
   98  cd
   99  ls
  100  python trc.py
  101  su -mn1
  102  su - mn1
  103  updatedb
  104  locate terracoin.conf
  105  terracoin-cli --help
  106  terracoin-cli -rcpuser
  107  terracoin-cli --help | more
  108  crontab -l -u mn1
  109  cd /usr/local/bin/terracoin
  110  cd /usr/local/bin/terracoind
  111  ls /usr/local/bin/terracoind
  113  locate terra|more
  114  cd /home/mn1/.terracoincore/
  115  ls
  116  more masternode.conf
  117  more terracoin.conf
  121  terracoin-cli --help | more
  122  terracoin-cli getinfo
  124  su - mn1 terracoin-cli getinfo
  126  updatedb
  127  locate terra
  128  terracoin-cli getinfo
  129  killall terracoind
  130  terracoind -printtoconsole
  131  shutdown -fr now

The last time I deleted the mn1 user appears to be entry 73. I think it was probably working by entry number 125. So, it looks like lines 77-88 are where I copied and pasted from clockuniverse's guide and found success.

Thanks again for the script!
